esdt-safe/src/lib.rs

```diff
@@ -349,76 +358,83 @@ pub trait EsdtSafe:
-                transaction_details.actual_bridged_amount,
-                transaction_details.required_fee,
-                transaction_details.refund_info.initial_batch_id,
-                transaction_details.refund_info.initial_nonce,
-            );
-        }
+        let transaction_details =
+            self.create_transaction_common(to, BigUint::zero(), opt_refund_info);
+
+        self.create_refund_transaction_event(
+            transaction_details.batch_id,
+            transaction_details.tx_nonce,
+            transaction_details.payment_token,
+            transaction_details.actual_bridged_amount,
+            transaction_details.required_fee,
+            transaction_details.refund_info.initial_batch_id,
+            transaction_details.refund_info.initial_nonce,
+        );
     }
 
     #[payable("*")]
-    #[endpoint(createTransactionSCCall)]
-    fn create_transaction_sc_call(
+    #[endpoint(createRefundTransactionSCCall)]
+    fn create_refund_transaction_sc_call(
         &self,
         to: EthAddress<Self::Api>,
         data: ManagedBuffer<Self::Api>,
         opt_refund_info: OptionalValue<RefundInfo<Self::Api>>,
     ) {
-        let transaction_details = self.create_transaction_common(to, opt_refund_info);
-
-        if !transaction_details.is_refund_tx {
-            self.create_transaction_sc_call_event(
-                transaction_details.batch_id,
-                transaction_details.tx_nonce,
-                transaction_details.payment_token,
-                transaction_details.actual_bridged_amount,
-                transaction_details.required_fee,
-                transaction_details
-                    .refund_info
-                    .address
-                    .as_managed_buffer()
-                    .clone(),
-                transaction_details.to_address,
-                data,
-            );
-        } else {
-            self.create_refund_transaction_sc_call_event(
-                transaction_details.batch_id,
-                transaction_details.tx_nonce,
-                transaction_details.payment_token,
-                transaction_details.actual_bridged_amount,
-                transaction_details.required_fee,
-                transaction_details.refund_info.initial_batch_id,
-                transaction_details.refund_info.initial_nonce,
-                data,
-            );
-        }
+        let transaction_details =
+            self.create_transaction_common(to, BigUint::zero(), opt_refund_info);
+
+        self.create_refund_transaction_sc_call_event(
+            transaction_details.batch_id,
+            transaction_details.tx_nonce,
+            transaction_details.payment_token,
+            transaction_details.actual_bridged_amount,
+            transaction_details.required_fee,
+            transaction_details.refund_info.initial_batch_id,
+            transaction_details.refund_info.initial_nonce,
+            data,
+        );
```

Comment:
wouldn't it be better to be included in create_refund_transaction endpoint and include a condition based on which different events are emitted? or does this have another purpose?

Reply:
Completely removed `create_refund_transaction_sc_call`. No longer needed in v3.1